deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -427,4 +434,117 @@ private void deltaWait(long milliSeconds) {
+     * 3. Else return replica info for all pods
+     * </p>
+     */
+    @Override
+    public List<ReplicaInfo> getReplicas(AuthConfig authConfig, String appName, String serviceName, String namespace, 
+            boolean isFilter) throws HyscaleException {
+        V1PodHandler v1PodHandler = (V1PodHandler) ResourceHandlers.getHandlerOf(ResourceKind.POD.getKind());
+        String selector = ResourceSelectorUtil.getServiceSelector(appName, serviceName);
+        ApiClient apiClient = clientProvider.get((K8sAuthorisation) authConfig);
+        List<V1Pod> podList = v1PodHandler.getBySelector(apiClient, selector, true, namespace);
+        
+        if (podList == null || podList.isEmpty()) {
+            return null;
+        }
+        
+        if (!isFilter) {
+            return K8sReplicaUtil.getReplicaInfo(podList);
+        }
+        
+        if (!PodPredicates.isPodAmbiguous().test(podList)) {
+            return K8sReplicaUtil.getReplicaInfo(podList);
+        }
+        String podOwner = K8sPodUtil.getCommonPodsOwner(podList);
+        ResourceKind podOwnerKind = ResourceKind.fromString(podOwner);
+        
+        // Unknown parent
+        if (StringUtils.isBlank(podOwner)) {
+            logger.debug("Unable to determine latest deployment, displaying all replicas");
+            WorkflowLogger.warn(DeployerActivity.LATEST_DEPLOYMENT_NOT_IDENTIFIABLE);
+            return K8sReplicaUtil.getReplicaInfo(podList);
+        }
+        
+        // Deployment
+        if (ResourceKind.REPLICA_SET.equals(podOwnerKind) || ResourceKind.DEPLOYMENT.equals(podOwnerKind)) {
+            // Get deployment, get revision, get RS with the revision, get all labels and filter pods
+            return K8sReplicaUtil.getReplicaInfo(filterPodsByDeployment(apiClient, appName, serviceName, namespace, podList));
+        }
+        
+        // TODO do we need to handle STS cases ??
+        logger.debug("Replicas info:: unhandled case, pod owner: {}", podOwner);
+        return K8sReplicaUtil.getReplicaInfo(podList);
+    }
+    
+    /**
+     * Get {@link V1Deployment} from cluster based on namespace, appname and service name.
+     * Deployment provides the revision for corresponding replica set
+     * Get {@link V1ReplicaSet} from cluster based on namespace, appname, service name and revision.
+     * Replica set provides pod-template-hash label(cluster internal) for corresponding pods
+     * From the provided pods return the ones which contains pod-template-hash in label 
+     * 
+     * @param apiClient
+     * @param appName
+     * @param serviceName
+     * @param namespace
+     * @param podList
+     * @return pods from pod list which refer to deployment for the app and service in namespace
+     */
+    public List<V1Pod> filterPodsByDeployment(ApiClient apiClient, String appName, String serviceName, String namespace, List<V1Pod> podList){
+        String selector = ResourceSelectorUtil.getServiceSelector(appName, serviceName);
```

Comment:
KubernetesDeployer should not expose this api outside instead this should be some other utility service . Also , if we can add a predicate to filter pods , so pods are always filtered by the predicate.